### PR TITLE
upgrade spring-boot and related dependencies

### PR DESCRIPTION
**Pull request by snopoke**
I tested this locally but I'm going to put it on staging for a bit just to be safe.

Dependency | Old version | New version
-- | -- | --
com.fasterxml:classmate | 1.5.0 | 1.5.1
com.fasterxml.jackson.core:jackson-annotations | 2.10.0 | 2.11.4
com.fasterxml.jackson.core:jackson-core | 2.10.0 | 2.11.4
com.fasterxml.jackson.core:jackson-databind | 2.10.0 | 2.11.4
com.fasterxml.jackson.datatype:jackson-datatype-jdk8 | 2.10.0 | 2.11.4
com.fasterxml.jackson.datatype:jackson-datatype-jsr310 | 2.10.0 | 2.11.4
com.fasterxml.jackson.module:jackson-module-parameter-names | 2.10.0 | 2.11.4
com.github.ben-manes.caffeine:caffeine | 2.8.0 | 2.8.8
com.google.errorprone:error_prone_annotations | 2.3.3 | 2.3.4
com.jayway.jsonpath:json-path | 2.5.0 | 2.4.0
com.jayway.jsonpath:json-path-assert | 2.5.0 | 2.4.0
com.sun.activation:jakarta.activation |   | 1.2.2
com.sun.istack:istack-commons-runtime | 3.0.8 | 3.0.11
com.sun.xml.fastinfoset:FastInfoset | 1.2.16 |  
com.zaxxer:HikariCP | 3.4.1 | 3.4.5
commons-codec:commons-codec | 1.13 | 1.15
io.micrometer:micrometer-core | 1.3.0 | 1.6.3
io.micrometer:micrometer-registry-statsd | 1.3.0 | 1.6.3
io.projectreactor:reactor-core | 3.3.0.RELEASE | 3.4.2
jakarta.activation:jakarta.activation-api | 1.2.1 | 1.2.2
jakarta.validation:jakarta.validation-api | 2.0.1 |  
jakarta.xml.bind:jakarta.xml.bind-api | 2.3.2 | 2.3.3
net.bytebuddy:byte-buddy | 1.10.1 | 1.10.19
net.bytebuddy:byte-buddy-agent | 1.10.1 | 1.10.19
org.apache.commons:commons-lang3 | 3.9 | 3.11
org.apache.commons:commons-pool2 | 2.7.0 | 2.9.0
org.apache.directory.studio:org.apache.commons.io | 2.4 |  
org.apache.httpcomponents:httpclient | 4.5.10 | 4.5.13
org.apache.httpcomponents:httpcore | 4.4.12 | 4.4.14
org.apache.logging.log4j:log4j-api | 2.12.1 | 2.13.3
org.apache.logging.log4j:log4j-to-slf4j | 2.12.1 | 2.13.3
org.apache.tomcat.embed:tomcat-embed-core | 9.0.27 | 9.0.41
org.apache.tomcat.embed:tomcat-embed-el | 9.0.27 |  
org.apache.tomcat.embed:tomcat-embed-websocket | 9.0.27 | 9.0.41
org.aspectj:aspectjrt | 1.9.4 | 1.9.6
org.aspectj:aspectjweaver | 1.9.4 | 1.9.6
org.assertj:assertj-core | 3.13.2 | 3.18.1
org.checkerframework:checker-qual | 2.10.0 | 3.5.0
org.dom4j:dom4j | 2.1.1 | 2.1.3
org.flywaydb:flyway-core | 6.0.6 | 7.1.1
org.glassfish:jakarta.el |   | 3.0.3
org.glassfish.jaxb:jaxb-runtime | 2.3.2 | 2.3.3
org.glassfish.jaxb:txw2 | 2.3.2 | 2.3.3
org.hamcrest:hamcrest | 2.1 | 2.2
org.hamcrest:hamcrest-core |   | 2.2
org.hamcrest:hamcrest-library |   | 2.2
org.hdrhistogram:HdrHistogram | 2.1.11 | 2.1.12
org.hibernate:hibernate-core | 5.4.6.Final | 5.4.27.Final
org.hibernate.common:hibernate-commons-annotations | 5.1.0.Final | 5.1.2.Final
org.hibernate.validator:hibernate-validator | 6.0.17.Final |  
org.javassist:javassist | 3.24.0-GA | 3.27.0-GA
org.jboss:jandex | 2.0.5.Final | 2.1.3.Final
org.junit.jupiter:junit-jupiter | 5.5.2 | 5.7.0
org.junit.jupiter:junit-jupiter-api | 5.5.2 | 5.7.0
org.junit.jupiter:junit-jupiter-engine | 5.5.2 | 5.7.0
org.junit.jupiter:junit-jupiter-params | 5.5.2 | 5.7.0
org.junit.platform:junit-platform-commons | 1.5.2 | 1.7.0
org.junit.platform:junit-platform-engine | 1.5.2 | 1.7.0
org.jvnet.staxex:stax-ex | 1.8.1 |  
org.mockito:mockito-core | 3.7.7 | 3.6.28
org.mockito:mockito-inline | 3.7.7 | 3.6.28
org.mockito:mockito-junit-jupiter | 3.7.7 | 3.6.28
org.postgresql:postgresql | 42.2.8 | 42.2.18
org.slf4j:jul-to-slf4j | 1.7.28 | 1.7.30
org.slf4j:slf4j-api | 1.7.28 | 1.7.30
org.springframework:spring-aop | 5.2.0.RELEASE | 5.3.3
org.springframework:spring-aspects | 5.2.0.RELEASE | 5.3.3
org.springframework:spring-beans | 5.2.0.RELEASE | 5.3.3
org.springframework:spring-context | 5.2.0.RELEASE | 5.3.3
org.springframework:spring-context-support | 5.2.0.RELEASE | 5.3.3
org.springframework:spring-core | 5.2.0.RELEASE | 5.3.3
org.springframework:spring-expression | 5.2.0.RELEASE | 5.3.3
org.springframework:spring-jcl | 5.2.0.RELEASE | 5.3.3
org.springframework:spring-jdbc | 5.2.0.RELEASE | 5.3.3
org.springframework:spring-messaging | 5.2.0.RELEASE | 5.3.3
org.springframework:spring-orm | 5.2.0.RELEASE | 5.3.3
org.springframework:spring-oxm | 5.2.0.RELEASE | 5.3.3
org.springframework:spring-test | 5.2.0.RELEASE | 5.3.3
org.springframework:spring-tx | 5.2.0.RELEASE | 5.3.3
org.springframework:spring-web | 5.2.0.RELEASE | 5.3.3
org.springframework:spring-webmvc | 5.2.0.RELEASE | 5.3.3
org.springframework.boot:spring-boot | 2.2.0.RELEASE | 2.4.2
org.springframework.boot:spring-boot-actuator | 2.2.0.RELEASE | 2.4.2
org.springframework.boot:spring-boot-actuator-autoconfigure | 2.2.0.RELEASE | 2.4.2
org.springframework.boot:spring-boot-autoconfigure | 2.2.0.RELEASE | 2.4.2
org.springframework.boot:spring-boot-devtools | 2.2.0.RELEASE | 2.4.2
org.springframework.boot:spring-boot-starter | 2.2.0.RELEASE | 2.4.2
org.springframework.boot:spring-boot-starter-actuator | 2.2.0.RELEASE | 2.4.2
org.springframework.boot:spring-boot-starter-aop | 2.2.0.RELEASE | 2.4.2
org.springframework.boot:spring-boot-starter-cache | 2.2.0.RELEASE | 2.4.2
org.springframework.boot:spring-boot-starter-data-jpa | 2.2.0.RELEASE | 2.4.2
org.springframework.boot:spring-boot-starter-jdbc | 2.2.0.RELEASE | 2.4.2
org.springframework.boot:spring-boot-starter-json | 2.2.0.RELEASE | 2.4.2
org.springframework.boot:spring-boot-starter-logging | 2.2.0.RELEASE | 2.4.2
org.springframework.boot:spring-boot-starter-test | 2.2.0.RELEASE | 2.4.2
org.springframework.boot:spring-boot-starter-tomcat | 2.2.0.RELEASE | 2.4.2
org.springframework.boot:spring-boot-starter-validation | 2.2.0.RELEASE |  
org.springframework.boot:spring-boot-starter-web | 2.2.0.RELEASE | 2.4.2
org.springframework.boot:spring-boot-test | 2.2.0.RELEASE | 2.4.2
org.springframework.boot:spring-boot-test-autoconfigure | 2.2.0.RELEASE | 2.4.2
org.springframework.data:spring-data-commons | 2.2.0.RELEASE | 2.4.3
org.springframework.data:spring-data-jpa | 2.2.0.RELEASE | 2.4.3
org.springframework.data:spring-data-keyvalue | 2.2.0.RELEASE | 2.4.3
org.springframework.data:spring-data-redis | 2.2.0.RELEASE | 2.4.3
org.springframework.integration:spring-integration-core | 5.2.0.RELEASE | 5.4.3
org.springframework.integration:spring-integration-redis | 5.2.0.RELEASE | 5.4.3
org.springframework.retry:spring-retry | 1.2.4.RELEASE | 1.3.1
org.xerial:sqlite-jdbc | 3.28.0 | 3.32.3.3
org.xmlunit:xmlunit-core | 2.6.3 | 2.7.0
org.yaml:snakeyaml | 1.25 | 1.27
redis.clients:jedis | 3.1.0 | 3.3.0

### CODE REVIEW

File: src/main/java/org/commcare/formplayer/application/WebAppContext.java [hunk 1]
```diff
@@ -107,11 +107,6 @@
     @Value("${detailed_tagging.tag_names:}")
     private List<String> detailedTagNames;
 
-    @Override
-    public void configureDefaultServletHandling(DefaultServletHandlerConfigurer configurer) {
-        configurer.enable();
-    }
```
Comment (snopoke) [inline src/main/java/org/commcare/formplayer/application/WebAppContext.java:113]: this is only needed for serving static dependencies and errors in the new version version

**Review by orangejenny — APPROVED** [review]
Do you think it's worth also having the QA team run a formplayer regression?

**Comment by snopoke** [comment]
> Do you think it's worth also having the QA team run a formplayer regression?

probably

**Comment by snopoke** [comment]
QA ticket here: https://dimagi-dev.atlassian.net/browse/QA-2616

**Pull request merged**